src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -172,6 +177,25 @@ public override void ProjectRemoved(Project underlyingProject)
             }
         }
 
+        public override void ProjectBuildComplete(Project underlyingProject)
+        {
+            if (underlyingProject == null)
+            {
+                throw new ArgumentNullException(nameof(underlyingProject));
+            }
+
+            if (_projects.TryGetValue(underlyingProject.Id, out var original))
+            {
+                // Doing an update to the project should keep computed values, but mark the project as dirty if the
+                // underlying project is newer.
+                var snapshot = original.WithProjectChange(underlyingProject);
+                _projects[underlyingProject.Id] = snapshot;
+
+                // Notify the background worker so it can trigger tag helper discovery.
```

Comment:
Leave a comment here as to why we don't check is dirty to remind our future selves.